include/deal.II/non_matching/mapping_info.h

```diff
@@ -0,0 +1,230 @@
+     */
+    dealii::internal::FEValuesImplementation::MappingRelatedData<dim, spacedim>
+      mapping_data;
+  };
+
+  // ----------------------- template functions ----------------------
+
+
+  template <int dim, int spacedim>
+  MappingInfo<dim, spacedim>::MappingInfo(const Mapping<dim> &mapping,
+                                          const UpdateFlags   update_flags)
+    : mapping(&mapping)
+  {
+    update_flags_mapping = update_default;
+    // translate update flags
+    if (update_flags & update_jacobians)
+      update_flags_mapping |= update_jacobians;
+    if (update_flags & update_gradients ||
+        update_flags & update_inverse_jacobians)
+      update_flags_mapping |= update_inverse_jacobians;
+
+    // always save quadrature points for now
+    update_flags_mapping |= update_quadrature_points;
+  }
+
+
+
+  template <int dim, int spacedim>
+  void
+  MappingInfo<dim, spacedim>::reinit(
+    const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+    const ArrayView<const Point<dim>> &                         unit_points)
+  {
+    this->unit_points =
+      std::vector<Point<dim>>(unit_points.begin(), unit_points.end());
+    compute_mapping_data_for_generic_points(cell, unit_points);
+  }
+
+
+
+  template <int dim, int spacedim>
+  const std::vector<Point<dim>> &
+  MappingInfo<dim, spacedim>::get_unit_points() const
+  {
+    return unit_points;
+  }
+
+
+
+  template <int dim, int spacedim>
+  const dealii::internal::FEValuesImplementation::MappingRelatedData<dim,
+                                                                     spacedim> &
+  MappingInfo<dim, spacedim>::get_mapping_data() const
+  {
+    return mapping_data;
+  }
+
+
+
+  template <int dim, int spacedim>
+  const Mapping<dim, spacedim> &
+  MappingInfo<dim, spacedim>::get_mapping() const
+  {
+    return *mapping;
+  }
+
+
+
+  template <int dim, int spacedim>
+  void
+  MappingInfo<dim, spacedim>::compute_mapping_data_for_generic_points(
+    const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+    const ArrayView<const Point<dim>> &                         unit_points)
+  {
+    if (const MappingQ<dim, spacedim> *mapping_q =
+          dynamic_cast<const MappingQ<dim, spacedim> *>(&(*mapping)))
+      {
+        mapping_q->fill_mapping_data_for_generic_points(cell,
+                                                        unit_points,
+                                                        update_flags_mapping,
+                                                        mapping_data);
+      }
+    else if (const MappingCartesian<dim, spacedim> *mapping_cartesian =
+               dynamic_cast<const MappingCartesian<dim, spacedim> *>(
+                 &(*mapping)))
+      {
+        mapping_cartesian->fill_mapping_data_for_generic_points(
+          cell, unit_points, update_flags_mapping, mapping_data);
+      }
+    else
+      {
+        FE_DGQ<dim, spacedim>   dummy_fe(1);
+        FEValues<dim, spacedim> fe_values(
```

Comment:
Since we're inside the `nonmatching` namespace that also provides an `FEValues` object, you need to qualify this with the deal.II main namespace:
```suggestion
        dealii::FEValues<dim, spacedim> fe_values(
```